Modules/_datetimemodule.c

```diff
@@ -1631,32 +1632,42 @@ wrap_strftime(PyObject *object, PyObject *format, PyObject *timetuple,
             ntoappend = 1;
         }
         /* A % has been seen and ch is the character after it. */
-        else if (ch == 'z') {
-            if (zreplacement == NULL) {
+        else if (ch == 'z' || (ch == ':' && *pin == 'z' && pin++)) {
```

Comment:
Is it still worth trying to share code between `%z` and `%:z`? Maybe duplicating the code with changes makes things simpler in the end.

Reply:
ok